CalendarApp.java

```diff
@@ -0,0 +1,419 @@
+        while (choice_exist) {
+            choice = scanner.nextLine();
+            switch (choice){
+                case "1":  //event is Meeting
+                    //get Date of Meeting
+                    System.out.println("Enter Event Details:");
+                    System.out.println("Enter Day(1-30):");
+                     Day=scanner.nextInt();
+                  if(Day<1||Day>30) {
+                      System.out.println("Not in range. Day was set automatically to 1.");
+
+                  }
+                    System.out.println("Enter Hour:");
+                     Hour=scanner.nextInt();
+                    System.out.println("Enter Minutes:");
+                     Minutes=scanner.nextInt();
+
+
+                     d=new Date(2023,6,Day,Hour,Minutes);
+                    //get duration from user
+                    System.out.println("Enter Duration of the event:");
+                    System.out.println("notice Duration can range from 1-60 in minutes.");
+                     duration =scanner.nextInt();
+
+                    if(duration>60||duration<1)
+                    {
+                        duration=30;
+                        System.out.println("Not in range. duration was set automatically to 30 minutes");
+
+                    }
+
+                    //get contact from user
+                    System.out.println("Enter Contact Name for meeting:");
+                    String contactName=scanner.nextLine();
+                    contactName=scanner.nextLine();
+                    //check if contact exists
+                    Contact contact=phoneBook.searchContactByName(contactName);
+                    if(contact!=null)
+                    {
+                        Event e1=new Event(d,duration);
+                        Event meeting=new Meeting(e1,contact);
+                        int index=d.getDate();
+                        this._dailyCalendar[index-1].add(meeting);//insert Meeting to RELEVANT DAY
+                        System.out.println("Meeting with "+contact.getName()+"inserted to calendar");
+                        this.SortDateArraylist(Day-1);
+
+                    }
+                    else{
+                        System.out.println("Contact does not exist, Meeting is not created");
+                    }
+
+                    choice_exist=false;
+                    break;
+
+                case "2": //Event is Occurrence
+                    System.out.println("Enter Event Details:");
+                    System.out.println("Enter Day(1-30):");
+                     Day=scanner.nextInt();
+                    if(Day<1||Day>30) {
+                        System.out.println("Not in range. Day was set automatically to 1.");
+
+                    }
+                    System.out.println("Enter Hour:");
+                     Hour=scanner.nextInt();
+                    System.out.println("Enter Minutes:");
+                     Minutes=scanner.nextInt();

```

Comment:
Line 90-93
there is no checks for out of range Hours (can enter 25 for example) or Minutes (can enter 399 for example)